eng/tools/TypeSpecValidation/src/rules/compile.ts

```diff
@@ -13,22 +13,24 @@ export class CompileRule implements Rule {
     let errorOutput: string | undefined;
 
     if (await checkFileExists(path.join(folder, "main.tsp"))) {
-      let [std, err] = await runCmd(`npx --no tsp compile . --warn-as-error`, folder);
-      stdOutput += std;
-      if (err == null) {
+      let [err, stdout, stderr] = await runCmd(`npx --no tsp compile . --warn-as-error`, folder);
+      stdOutput += stdout;
+      if (err) {
         success = false;
-        errorOutput += err;
+        errorOutput += err.message;
+        errorOutput += stderr;
       }
     }
     if (await checkFileExists(path.join(folder, "client.tsp"))) {
-      let [std, err] = await runCmd(
+      let [err, stdout, stderr] = await runCmd(
         `npx --no tsp compile client.tsp --no-emit --warn-as-error`,
         folder
       );
-      stdOutput += std;
-      if (err == null) {
+      stdOutput += stdout;
+      if (err) {
         success = false;
-        errorOutput += err;
+        errorOutput += err.message;
+        errorOutput += stderr;
```

Comment:
```suggestion
        errorOutput += stderr;
        errorOutput += err.message;
```

In a console scenario, I believe you would see `stderr` first before `err.message`, so let's do the same.